transport/tchannel/transport.go

```diff
@@ -333,22 +338,44 @@ func (t *Transport) onPeerStatusChanged(tp *tchannel.Peer) {
 // CreateTLSOutboundChannel creates a outbound channel for managing tls
 // connections with the given tls config and destination name.
 // Usage:
-// 	tr, _ := tchannel.NewTransport(...)
-//  outboundCh, _ := tr.CreateTLSOutboundChannel(tls-config, "dest-name")
-//  outbound := tr.NewOutbound(peer.NewSingle(id, outboundCh))
+//
+//		tr, _ := tchannel.NewTransport(...)
+//	 outboundCh, _ := tr.CreateTLSOutboundChannel(tls-config, "dest-name")
+//	 outbound := tr.NewOutbound(peer.NewSingle(id, outboundCh))
 func (t *Transport) CreateTLSOutboundChannel(tlsConfig *tls.Config, destinationName string) (peer.Transport, error) {
+	dialerFunc := t.dialer
+	if t.enableMPTCP {
+		dialerFunc = dialMPTCPContext
+	}
 	params := dialer.Params{
 		Config:        tlsConfig,
 		Meter:         t.meter,
 		Logger:        t.logger,
 		ServiceName:   t.name,
 		TransportName: TransportName,
 		Dest:          destinationName,
-		Dialer:        t.dialer,
+		Dialer:        dialerFunc,
```

Comment:
mTLS and MPTCP is a bit nested here, I'm not sure if there is better way to implement it.

Reply:
@jronak to confirm. But from my reading, the dialer is used to create an outbound TCP connection and your change is just doing that by conditionally passing on old dialer or mptcp dialer. mTLS is [AFTER your established the TCP connection](https://pkg.go.dev/crypto/tls#Client):
> Client returns a new TLS client side connection using conn as the underlying transport. 

Your change is good.

Reply:
On my understanding, this function is used in [buildUnaryOutbound:getPeerTransport](https://github.com/yarpc/yarpc-go/blob/dev/transport/tchannel/config.go#L212). `buildUnaryOutbound` builds unary outbounds from parsed configuration.

Reply:
If `t.dialer ` is passed in by options, those lines of code will overwrite user-passed in dialer into MPTCP dialer. Maybe we should just change dialer when [buildTransport](https://github.com/yarpc/yarpc-go/blob/dev/transport/tchannel/config.go#L156)?  [buildTransport](https://github.com/yarpc/yarpc-go/blob/dev/transport/tchannel/config.go#L156) will change `t.dialer ` directly, so we don't need to do it in `buildUnaryOutbound`.